specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -472,71 +417,75 @@ namespace Microsoft.Windows.PushNotifications
     };
 
     // The PushNotificationChannel Progress result
+    [feature(Feature_PushNotifications)]
     struct PushNotificationCreateChannelStatus
     {
-        // The last extended error we failed Channel requests on that caused the inprogress retry status. S_OK if this is just progress status.
-        HRESULT extendedError;
-
         // Either InProgress or InProgressRetry status
         PushNotificationChannelStatus status;
 
+        // The last extended error we failed Channel requests on that caused the inprogress retry status. E_PENDING if this is just progress status.
+        HRESULT extendedError;
+
         // Total Retries so far
         UInt32 retryCount;
     };
 
+    [feature(Feature_PushNotifications)]
     runtimeclass PushNotificationChannel
     {
-        PushNotificationChannel(String uri, Windows.Foundation.DateTime expiration);
-
         // The Channel Uri for app to Post a notification to.
-        String Uri { get; };
+        Windows.Foundation.Uri Uri { get; };
 
         // Expiration of the Channel
         Windows.Foundation.DateTime ExpirationTime { get; };
 
         // Unsubscribes the channel
         void Close();
-
-        // In-memory Event handler for Push Notifications
-        event Windows.Foundation.TypedEventHandler<PushNotificationChannel, PushNotificationReceivedEventArgs> PushReceived;
     }
 
+    [feature(Feature_PushNotifications)]
     runtimeclass PushNotificationCreateChannelResult
     {
         PushNotificationCreateChannelResult(
             PushNotificationChannel channel,
             HRESULT extendedError,
             PushNotificationChannelStatus status);
 
-        // The Push channel associated with the Result. Null if InProgress or completion failed
+        // The Push channel associated with the Result. Valid only if status is CompletedSuccess.
         PushNotificationChannel Channel { get; };
 
         // More detailed error code in addition to the ChannelStatus state.
         HRESULT ExtendedError{ get; };
 
-        // The Status of the ChannelComplete operation
+        // The Status of the CreateChannel operation
         PushNotificationChannelStatus Status { get; };
     };
 
-    static runtimeclass PushNotificationManager
+    [feature(Feature_PushNotifications)]
+    runtimeclass PushNotificationManager
```

Comment:
I wonder if the model should be more like Register/Start/Stop/Unregister, where:

* Register configures all the registration goo, but does not start the flow of notifications yet
* Start begins really listening and handing incoming pushes to the handler
* Stop ends the flow of notifications to this manager instance so the app can shut down
* Unregister removes all channels that were configured and is basically the reverse of Register

No change required, something to think about.

Reply:
https://github.com/microsoft/WindowsAppSDK/issues/2227

Reply:
Tracking this.